tests/BuildScriptGenerator.Tests/DotnetCore/GlobalJsonSdkResolverTest.cs

```diff
@@ -14,217 +11,151 @@ namespace Microsoft.Oryx.BuildScriptGenerator.Tests.DotNetCore
-        [InlineData("3.1.200-preview1-014995", "3.1.200-preview1-014996", -1)]
-        [InlineData("5.0.100-preview.1.20155.7", "5.0.100-preview.1.20155.7", 0)]
-        [InlineData("5.0.100-preview.1.20155.7", "5.0.100-preview.2.20155.7", -1)]
-        [InlineData("5.0.100-preview.1.20155.7", "5.0.100-preview.1.20155.8", -1)]
-        public void PreviewVersions_AreCompared_UsingStringComparisionRules(
-            string previewVersion,
-            string nonPreviewVersion,
-            int expected)
+        [InlineData("AllowPrerelease")]
+        [InlineData("allowPrerelease")]
+        [InlineData("allowprerelease")]
+        public void GlobalJsonDeserialization_IsCaseInsensitiveForAllowPrereleaseKey(string allowPrereleaseKey)
         {
             // Arrange
-            var previewVersionInfo = SdkVersionInfo.Parse(previewVersion);
-            var nonPreviewVersionInfo = SdkVersionInfo.Parse(nonPreviewVersion);
+            var globalJsonTemplate = @"
+            {
+                ""sdk"": {
+                    ""version"": ""3.1.100"",
+                    ""#allowPrerelease-key#"": true,
+                }
+            }";
+            var globalJson = globalJsonTemplate
+                .Replace("#allowPrerelease-key#", allowPrereleaseKey);
+            var sourceRepo = new MemorySourceRepo();
+            sourceRepo.AddFile(globalJson, DotNetCoreConstants.GlobalJsonFileName);
+            var expectedVersion = "3.1.102-preview2-03444";
+            var availableVersions = new[] { "3.1.100", "3.1.101", "3.1.102-preview1-03444", "3.1.102-preview2-03444" };
+            var runtimeVersion = "2.1.12";
+            var globalJsonHelper = GetGlobalJsonHelper();
 
             // Act
-            var actual = previewVersionInfo.CompareTo(nonPreviewVersionInfo);
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(sourceRepo, runtimeVersion, availableVersions);
 
             // Assert
-            Assert.Equal(expected, actual);
+            Assert.Equal(expectedVersion, actual);
         }
 
         [Theory]
-        [InlineData("3.1.200-preview1-014995", "3.1.200-PreView1-014995", 0)]
-        [InlineData("3.1.200-preview1-014995", "3.1.200-PreView2-014995", -1)]
-        [InlineData("3.1.200-preview1-014995", "3.1.200-PreView1-014996", -1)]
-        [InlineData("5.0.100-preview.1.20155.7", "5.0.100-PreView.1.20155.7", 0)]
-        [InlineData("5.0.100-preview.1.20155.7", "5.0.100-PreView.2.20155.7", -1)]
-        [InlineData("5.0.100-preview.1.20155.7", "5.0.100-PreView.1.20155.8", -1)]
-        public void PreviewVersions_AreCompared_CaseInsensitively(
-            string previewVersion,
-            string nonPreviewVersion,
-            int expected)
+        [InlineData("3.0.3", "3.2.201")]
+        [InlineData("2.1.16", "2.3.202")]
+        [InlineData("5.0.0", "5.2.100-rc1-01445")]
+        public void GetSatisfyingSdkVersion_MustReturnLatestMinorOfRuntimeVersion_IfGlobalJsonFileIsNotFoundInTheRepo(
```

Comment:
@arroyc This is the scenario we were discussing in my earlier PR related to global.json file not being present and what version should be choose to install. Here latest minor is used.